frontend/src/components/ui/WorkspaceTabs.jsx

```diff
@@ -27,9 +32,22 @@ export default function WorkspaceTabs() {
   const handleTabChange = (evt) => {
     const index = evt.selectedIndex;
     const key = renderedTabs[index]?.id;
+
+    setZoomLevels((prevZoomLevels) => ({
+      ...prevZoomLevels,
+      [workspace.active]: editor?.zoom,
+    }));
+
     setWorkspace((draft) => {
       draft.active = key;
     });
+
+    if (editor) {
+      editor.zoom = zoom_levels[key] !== undefined ? zoom_levels[key] : 1;
+      editor.zoom_refresh(); // Refresh after setting zoom
+    }
+
+    console.log(editor.zoom);
```

Comment:
I don't think this log is necessary

Reply:
just forgot to remove this after debugging.